src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs

```diff
@@ -66,6 +66,13 @@ public class LocalizedTaxonomyController : Controller {
                         var firstTermForCulture = appliedTerms.FirstOrDefault(x => x.As<LocalizationPart>() != null && x.As<LocalizationPart>().Culture != null && x.As<LocalizationPart>().Culture.Culture == culture);
                         if (firstTermForCulture != null) {
                             firstTermIdForCulture = firstTermForCulture.Id;
+                        } else {
+                            // If there is no valid localization, firstTermForCulture is null.
+                            // To avoid that, use the first checked term (if any is checked).
+                            var firstTerm = terms.FirstOrDefault(t => t.IsChecked = appliedTerms.Select(x => x.Id).Contains(t.Id));
```

Comment:
```suggestion
                            var firstTerm = terms.FirstOrDefault(t => t.IsChecked = appliedTerms.Any(x => x.Id == t.Id));
```

this should be simpler.

Also this whole FirstOrDefault call is complex, it's actually doing an assignment and stops at the first one that is true. Was that intentional?